src/Sylius/Component/Core/Promotion/Action/UnitFixedDiscountPromotionActionCommand.php

```diff
@@ -93,6 +92,8 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
         foreach ($filteredItems as $item) {
```

Comment:
And before that loop we can do `if(empty($filteredItems)) { return false; }` and have the logic working for filters also 🎉